extensions/smallrye-health/runtime/src/main/java/io/quarkus/smallrye/health/runtime/ShutdownReadinessCheck.java

```diff
@@ -0,0 +1,27 @@
+package io.quarkus.smallrye.health.runtime;
+
+import javax.inject.Singleton;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.Readiness;
+
+@Singleton
```

Comment:
One day I would understand why sometimes we use @ApplicationScoped, and sometimes @Singleton (I know it's not coming from the same spec, but still puzzled by when use which).

Reply:
ApplicationScoped requires a proxy to comply with the spec, Singleton is usually better, but because it is not actually in the CDI spec a lot of people just default to application scoped.